main_plot.py

```diff
@@ -148,23 +246,62 @@ def filter_dataframe_by_dataset_method_and_subset(
     ~filtered_dataframe.subset_row.isin(mask_low_r_squared)
 ]
 
-# decoding zip json
-filtered_dataframe_file_path_excl_cities = filtered_dataframe.loc[
-    filtered_dataframe["R2"] < -10
-].drop_duplicates(subset=["subset_row"])["Filepath"]
-print(f"cities with R2 <-10\n{filtered_dataframe_file_path_excl_cities}")
 
-with gzip.open(filtered_dataframe_file_path_excl_cities[4], "rt") as zipfile:
-    my_object = json.load(zipfile)
-print(my_object.keys())
-my_object_values = my_object["values"]
-print(my_object_values)
-print(type(my_object_values))
+def load_dataset(dataset_name: str) -> list[Dataset]:
+    """
+    Load the given dataset.
+    """
+    if (
+        dataset_name not in __dataset_row_items
+        or dataset_name == "airline_passengers"
+        or dataset_name == "sun_spots"
+        or dataset_name == "list_of_tuples"
+    ):
+        return [__dataset_loaders[dataset_name]()]
+    else:
+        return __dataset_loaders[dataset_name](__dataset_row_items[dataset_name])
+
```

Comment:
Try this (thanks ChatGPT):
```
exception_datasets = {
    "airline_passengers": None,
    "sun_spots": None,
    "list_of_tuples": None
}

if dataset_name in __dataset_row_items or dataset_name in exception_datasets:
    return __dataset_loaders[dataset_name](__dataset_row_items.get(dataset_name))
else:
    return [__dataset_loaders[dataset_name]()]
```

Reply:
Also we might want to document the function:
```
def load_dataset(dataset_name: str) -> list[Dataset]:
    """
    Load the given dataset.

    Args:
        dataset_name: A string representing the name of the dataset to load.

    Returns:
        A list of Dataset objects representing the loaded dataset.

    Raises:
        KeyError: If `dataset_name` is not a valid dataset name.

    Notes:
        If `dataset_name` is in `__dataset_row_items`, the corresponding dataset
        loader function is called with the row items (if any) from `__dataset_row_items`.
        Otherwise, the dataset loader function is called without any row items.
        The format of the row items is a dictionary with keys corresponding to the column
        names and values corresponding to the row values.
    """
    ...
```

Reply:
I think the above should be slightly corrected in the following way:

```

    if dataset_name in __dataset_row_items and dataset_name not in exception_datasets:
        return __dataset_loaders[dataset_name](__dataset_row_items.get(dataset_name))
    else:
        return [__dataset_loaders[dataset_name]()]
```